### DISCUSSION TITLE
Better results at low cfg indicate overtraining?

### DISCUSSION THREAD

**Discussion by Jonseed** [d8ahazard/sd_dreambooth_extension#449] [Q&A]
I trained a model that gives more realistic results if I set inferencing at a _very_ low cfg value, like 1-2. Does that indicate overtraining?

If so, I'm not sure how I overtrained, since I used the recommended number of images, steps, learning rate, and prior preservation class images. How did I overtrain? I did train the text encoder the whole time. Did that do it?

Here are the settings I used:

> SD1.x
> 44 images of a subject, a mix of expressions, clothing, lighting, etc
> detailed caption txt files
> 5100 steps
> 530 class images
> 0.00000172 learning rate
> train text encoder
> fp16 mixed precision
> use 8bit adam
> xformers memory attention
> don't cache latents
> NO train EMA
> gradient checkpointing

**Comment by manniacmind** [comment]
V1.5 or V2.0?
I don't know about the influence of the cfg value, but it could very well indicate that you overtrained. Try to put the concept name in [], [[]] or [[[]]] brackets with a cfg value of 7, to see if the results improve. This could indicate overtraining as well.

In v1.5 I had really good results with 16000 steps and a learning rate of 0,0000005 - in general lower learning rates help against overfitting. I also used 44 images, but 1500 class images. I also tried caption files, but I found that without caption files my results were actually better (maybe the extra words dilute the training and more steps at an even lower training rate would be needed?). I also found that training with EMA led to better results for me.

In V2.0 I was not able to get a good training result yet, even with the exact same settings as above.

**Reply by Jonseed** [reply]
It was a 1.x model that I was training on. Thanks for the tips. I'll try increasing my steps at a lower learning rate, and maybe triple my class images also. I'll try not using caption files too.

**Reply by Jonseed** [reply]
Unfortunately, I can't train with EMA, or I get OOM. Did you train the text encoder? I notice that some people only train the text encoder for a short time at the beginning and end of training. Maybe training the text encoder the whole time also overtrains?

**Reply by manniacmind** [reply]
I did train the text encoder, but didn't experiment with it. to be honest, most of the time I trial and error my way through :D

**Reply by mykeehu** [reply]
I don't know how much the text encoder is helpful or not, but for me it's just a checkbox that's always checked when I run a training. I can't turn it on at the beginning and end of a training unless I start it multiple times.
